### ISSUE TITLE
ES 6 Support?

### ISSUE THREAD

**Issue by LorenDorez** [Taritsyn/JavaScriptEngineSwitcher#95]
Hello,

We are currently using this with ReactJS.net. We use Webpack to compile all our JSX and then send the compiled JS to the JS engine.

We would like to remove support to IE11 (ES5) and natively support ES6 without having to bring in the bloat of polyfills to downgrade to ES5.

Is this possible with maybe chakra or V8 engines?

**Comment by Taritsyn** [comment]
Hello, Loren!

Yes, this is possible with the [ChakraCore](https://github.com/Taritsyn/JavaScriptEngineSwitcher/wiki/ChakraCore) and [V8](https://github.com/Taritsyn/JavaScriptEngineSwitcher/wiki/V8) modules.

If you use the V8 module, then you will have access to the most recent versions of ECMAScript.

**Comment by LorenDorez** [comment]
Awesome thanks going to work on getting this setup and tested

**Comment by Taritsyn** [comment]
It's worth noting that all of the above applies only to server-side rendering. Execution of client-side JS code will still depend on browsers.

**Comment by LorenDorez** [comment]
Yeah i have a small JS function i run for feature detection and then show a special message telling them their browser is unsupported. I just needed SSR to run as my special message is a compiled React component that i set 'display:none' on

**Comment by Taritsyn** [comment]
Then there should be no problems.
